localstack/services/sns/sns_listener.py

```diff
@@ -887,6 +931,81 @@ def get_message_attributes(req_data):
     return attributes
 
 
+def get_publish_batch_messages(req_data):
```

Comment:
I think we already have a way to parse the request messages that are using this format - can we try using [`parse_urlencoded_data(..)`](https://github.com/localstack/localstack/blob/10a51f8a09146125970ba49122704d6f1faf1985/localstack/utils/aws/aws_responses.py#L365-L377) here and see if it can be used for this particular use case? I'd prefer us to use the generic request parsing facilities (and extend it, if needed), rather than manually parsing the request as in this function.